canlink/mcap.go

```diff
@@ -0,0 +1,151 @@
+package canlink
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/foxglove/mcap/go/mcap"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"os"
+	"strconv"
+)
+
+type Mcap struct {
+	l               *zap.Logger
+	file            *os.File
+	createdChannels map[string]uint16
+	client          *CANClient
+	currIndex       uint16
+}
+
+func NewMcap(l *zap.Logger, c *CANClient, dir string, busName string) *Mcap {
+	f, err := createFile(l, dir, busName, ".mcap")
+	if err != nil {
+		l.Error(err.Error())
```

Comment:
Don't perform actions that could result in an error without returning the error.